content/articles/creating-a-hand-gesture-volume-controller-using-python-and-pycharm/index.md

```diff
@@ -0,0 +1,274 @@
+`lmList = []`: this code creates an *empty list* which will store the list of elements of the hands detected by the `mediapipe` hand module, i.e., the number of points on the hand.
+
+`if results.multi_hand_landmarks:` checks whether the input has multiple hands.
+
+### Creating a for loop to manipulate each hand
+```python
+for handlandmark in results.multi_hand_landmarks:
+    for id, lm in enumerate(handlandmark.landmark):
+        h, w, c = img.shape
+        cx, cy = int(lm.x * w), int(lm.y * h)
+        lmList.append([id, cx, cy]) 
+    mpDraw.draw_landmarks(img, handlandmark, mpHands.HAND_CONNECTIONS)
+```
+`for handlandmark in results.multi_hand_landmarks`: this creates a for loop to interact with each hand in the results.
+
+`for id, lm in enumerate(handlandmark.landmark)`: this is a for loop to get the `id` (id number) and `lm` (landmark information) for each handlandmark. The landmark information will give us the `x` and `y` coordinates. The id number is the number assigned to the various hand points.
+
+`h, w, c = img.shape`: this piece of code checks the height, width, and channels of our image. This will give us the width and height of the image.
+
+`cx, cy = int(lm.x * w), int(lm.y * h)`: we are finding the central position of our image. We achieve this by multiplying *lm.x by the width* and assigning the value obtained to `cx`. Then multiply `lm.y` by the height and assign the value obtained to `cy`. 
+
+> `lm` stands for landmark.
+
+`lmList.append([id, cx, cy])`: we are adding the values of `id`,`cx` and `cy` to `lmList`.
+
+` mpDraw.draw_landmarks(img, handlandmark, mpHands.HAND_CONNECTIONS)`: we pass our image (`img`) then we call *mpDraw.draw_landmarks* to draw all the landmarks of the hand.
+
+### Specifying the points of the thumb and middle finger we will use
+```python
+if lmList != []:
+    x1, y1 = lmList[4][1], lmList[4][2]
+    x2, y2 = lmList[8][1], lmList[8][2]
+```
+`if lmlist != []`: we use this statement to specify that the number of elements should not be null.
+ 
+`x1, y1 = lmList[4][1], lmList[4][2]` : we are assigning variables `x1` and `y1` the `x` and `y` coordinates of point `4` respectively. This is the tip of the thumb. 
+ 
+`x2, y2 = lmList[8][1], lmList[8][2]`: we are assigning variables `x2` and `y2` the `x` and `y` coordinates of point `8` respectively. This is the tip of the index finger. 
+ 
+Refer to the hand image diagram we discussed to identify the [points](#referral-hand-image).
+
+### Drawing a line between the tip of the thumb and the tip of the index finger
+```python                                            
+       cv2.circle(img, (x1, y1), 15, (255, 0, 0), cv2.FILLED)  
+       cv2.circle(img, (x2, y2), 15, (255, 0, 0), cv2.FILLED)  
+```
+`cv2.circle(img, (x1, y1), 4, (255, 0, 0), cv2.FILLED)`: we are calling the `cv2.circle` and passing our image `img`.
+
+` (x1, y1)`: it specifies that we will draw the circle at the tip of the thumb.
+
+`15`: this is the *radius* of the circle.

```

Comment:
As you can see, this section isn't really aesthetically pleasing to read as there are way too short statements. You can just give your explanations in paragraphs without repeating the code blocks.